.github/workflows/sync-upstream-change.yml

```diff
@@ -0,0 +1,94 @@
+name: Sync with Upstream (change)
+
+on:
+  pull_request_review:
+    types: [submitted]
+
+jobs:
+  sync:
+    runs-on: ubuntu-latest
+    if: github.event.review.state == 'changes_requested' && startsWith(github.event.pull_request.head.ref, 'sync-upstream-')
+    steps:
+      - name: Checkout the repository
+        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
+        with:
+          fetch-depth: 0
+          ref: ${{ github.event.pull_request.head.ref }}
+
+      - name: Reset branch to main
+        run: |
+          git reset --hard origin/main
+
+      - name: Configure Git
+        run: |
+          git config --local user.email "41898282+github-actions[bot]@users.noreply.github.com"
+          git config --local user.name "github-actions[bot]"
+
+      - name: Pull changes from upstream and merge
+        id: merge
+        run: |
+          git remote add upstream https://github.com/actions/runner-images.git
+          git fetch upstream
+          SHA=$(git rev-parse upstream/main)
+          echo "upstream-sha=$SHA" >> $GITHUB_OUTPUT
+          git merge $SHA || echo "conflicts=true" >> $GITHUB_OUTPUT
+
+      - name: Push changes to branch
+        run: |
+          git push -f origin ${{ github.event.pull_request.head.ref }}
+
+      - name: Set PR body
```

Comment:
Suggest putting the contents here into separate files since they're going to need to stay in sync between the two workflows. Can use `gomplate` or similar to do the substitution!

Reply:
I also noticed that, will look into what I can do!

Reply:
See https://github.com/grafana/runner-images/pull/7#discussion_r1647921226